### PR TITLE
Update how to check PCT and ATH tests

### PR DESCRIPTION
**Pull request by NotMyFault**
The change proposed updates the RC workflow a bit, to test the incremental _before_ the backporting PR is merged, to reduce the chance to introduce regressions.
Currently, we don't test the RC at all and rely on green checks on the PRs created by automatic tooling to update the lts lines _after_ the release.

While the chance is low, a PCT and ATH run can help to identify regressions early.

### CODE REVIEW

File: docs/releases.md [hunk 1]
```diff
@@ -194,11 +193,11 @@ Visit the checklist for more information.
 Additionally, take a look at the [release](https://github.com/jenkins-infra/release/issues?q=is%3Aclosed+label%3Alts-candidate+) and [packaging](https://github.com/jenkinsci/packaging/issues?q=is%3Aclosed+label%3Alts-candidate) repository, for additional LTS candidates.
 
 ### Review tests
-1. Review acceptance tests. Take a look at the checks of your backporting PR from step 10, and make sure, that `Tests / ath / Running ATH / ATH` are green:  
-![acceptance test screenshot](images/jenkins-ath.png)  
-The amount of tests may vary.
-2. Review BOM tests. Make sure, that all tests from your BOM PR are green.
-3. Review JCasC tests. Make sure, that all tests from your JCasC PR are green.
```
Comment (NotMyFault) [inline docs/releases.md:201]: JCasC is part of the BOM

File: docs/releases.md [hunk 2]
```diff
@@ -28,7 +28,6 @@ Bear in mind, as future release lead, you do not need to work through the releas
 * [Release Candidate Creation](#release-candidate-creation)
   * [⚠️ Merge the backporting PR](#merge-the-backporting-pr)
   * [⚠️ Publish a pre-release](#publish-a-pre-release)
-  * [Announce the release candidate](#send-an-announcement-email)
```
Comment (NotMyFault) [inline docs/releases.md:31]: That's covered by automatic tooling now

File: docs/releases.md [hunk 3]
```diff
@@ -194,11 +193,11 @@ Visit the checklist for more information.
...
 
 ### Review tests
-1. Review acceptance tests. Take a look at the checks of your backporting PR from step 10, and make sure, that `Tests / ath / Running ATH / ATH` are green:  
-![acceptance test screenshot](images/jenkins-ath.png)  
-The amount of tests may vary.
-2. Review BOM tests. Make sure, that all tests from your BOM PR are green.
-3. Review JCasC tests. Make sure, that all tests from your JCasC PR are green.
+The following tests need to be reviewed and confirmed to be passing:
+1. **Acceptance tests**: Open a pull request modifying the `jenkins.version` of the `lts` profile in the [acceptance test harness](https://github.com/jenkinsci/acceptance-test-harness/blob/ae19d1e9962f0b162cb3ef9033b24e9629d222c4/pom.xml#L778) suite, to the incremental produced by the backporting pull request in `jenkinsci/jenkins`.
+2. **Plugin compatibility**: Open a pull request modifying the `jenkins.version` in the newest [bom profile](https://github.com/jenkinsci/bom/blob/ce598171d115bfc9dc4d15e6f6c2bbb06e6fe7ed/sample-plugin/pom.xml#L889), to the incremental produced by the backporting pull request in `jenkinsci/jenkins`.
```
Comment (timja) [inline docs/releases.md:198]: it needs a file `full-test` created as well or labelled as `full-test` if you have permission:
https://github.com/jenkinsci/bom?tab=readme-ov-file#pct

Comment (NotMyFault) [reply docs/releases.md]: Good point, adding the file is the easiest solution, if you lack triage and write perms.

**Review by MarkEWaite — APPROVED** [review]
Thanks!

**Review by timja — APPROVED** [review]

**Pull request merged**